ocaml/testsuite/tests/typing-layouts/jkinds.ml

```diff
@@ -1565,3 +1565,104 @@ Error: This expression has type <  > but an expression was expected of type
+         of the definition of t at line 1, characters 0-33.
+       But the layout of type 'a must be a sublayout of value, because
+         of the definition of t at line 1, characters 0-33.
+|}]
+(* CR layouts v2.8: this should be accepted; 'a should be inferred to have kind
+   value mod global *)
+
+type 'a t : word = 'a
+[%%expect {|
+Line 1, characters 0-21:
+1 | type 'a t : word = 'a
+    ^^^^^^^^^^^^^^^^^^^^^
+Error: The layout of type 'a is value, because
+         of the definition of t at line 1, characters 0-21.
+       But the layout of type 'a must be a sublayout of word, because
+         of the definition of t at line 1, characters 0-21.
+|}]
+(* CR layouts v2.8: this should be accepted; 'a should be inferred to have kind
+   word *)
+
+type 'a t : any = private 'a
+[%%expect {|
+type 'a t = private 'a
+|}]
+
+type 'a t : value = private 'a
+[%%expect {|
+type 'a t = private 'a
+|}]
+
+type 'a t : value mod global = private 'a
+[%%expect {|
+Line 1, characters 0-41:
+1 | type 'a t : value mod global = private 'a
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The layout of type 'a is value, because
+         of the definition of t at line 1, characters 0-41.
+       But the layout of type 'a must be a sublayout of value, because
+         of the definition of t at line 1, characters 0-41.
+|}]
+(* CR layouts v2.8: this should be accepted; 'a should be inferred to have kind
+  value mod global *)
+
+type 'a t : word = private 'a
+[%%expect {|
+Line 1, characters 0-29:
+1 | type 'a t : word = private 'a
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The layout of type 'a is value, because
+         of the definition of t at line 1, characters 0-29.
+       But the layout of type 'a must be a sublayout of word, because
+         of the definition of t at line 1, characters 0-29.
+|}]
+(* CR layouts v2.8: this should be accepted; 'a should be inferred to have kind
+  word *)
+
+type 'a t : value mod global = Foo of 'a [@@unboxed]
+[%%expect {|
+Line 1, characters 0-52:
+1 | type 'a t : value mod global = Foo of 'a [@@unboxed]
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The layout of type t is value, because
+         it instantiates an unannotated type parameter of t, defaulted to layout value.
+       But the layout of type t must be a sublayout of value, because
+         of the annotation on the declaration of the type t.
+|}]
+(* CR layouts v2.8: this should be accepted; 'a should be inferred to have kind
+  value mod global *)
+
+type 'a t : value mod global = { x : 'a }
+[%%expect {|
+Line 1, characters 0-41:
+1 | type 'a t : value mod global = { x : 'a }
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The layout of type t is value, because
+         it's a boxed record type.
+       But the layout of type t must be a sublayout of value, because
+         of the annotation on the declaration of the type t.
+|}]
+(* CR layouts v2.8: this should be accepted; 'a should be inferred to have kind
```

Comment:
Good catch, I chose my modal axis poorly for this test. I should have chosen one of `portable`, `uncontended`, or `many` instead of `global` to test the behavior I had meant to. I've updated the test to  reflect this. Do you agree with it now?

Reply:
No, still disagree. I agree that if `'a` crosses `many`, then it's safe for the record to, also. But I don't yet expect to be able to infer that. Maybe once we give an initial kind of `immutable_data with 'a`, which is `value mod many uncontended portable with 'a`, and then check `value mod many uncontended portable with 'a  <:  value mod many`, and then be able to reduce that to `linearity_bound_of('a) <: many`, we could get this. But that's a long way off.

Reply:
ack, I've removed the CR from the `value mod many` test.